packages/@uppy/dashboard/src/components/FileList.js

```diff
@@ -32,22 +51,34 @@ module.exports = (props) => {
     cancelUpload: props.cancelUpload,
     toggleFileCard: props.toggleFileCard,
     removeFile: props.removeFile,
-    handleRequestThumbnail: props.handleRequestThumbnail
+    handleRequestThumbnail: props.handleRequestThumbnail,
+    handleCancelThumbnail: props.handleCancelThumbnail
   }
 
-  function renderItem (fileID) {
+  const rows = chunks(Object.keys(props.files), props.itemsPerRow)
+
+  function renderRow (row) {
     return (
-      <FileItem
-        key={fileID}
-        {...fileProps}
-        file={props.files[fileID]}
-      />
+      // Use the first file ID as the key—this should not change across rerenders
+      <div role="presentation" key={row[0]}>
```

Comment:
I think we should put all the `presentation`-roles and `list`-roles in one place, and add a comment about why we're doing this.

E.g.:
1. Move `<VirtualList role="list"`/> prop inside the `<VirtualList/>`.
2. Instead of `props.renderRow` - pass `props.renderItem` and `props.getRowKey`.
3. Add a comment in `VirtualList` about what `presentation` is supposed to be doing here.

What do you think?

Reply:
Cause it will keep related semantics together, and cause `presentation` really needs an explanation here (but it would be nice to hide that explanation into the `VirtualList`).

Reply:
hmm. Right now VirtualList only knows about windowing a uni-directional list, like the `preact-virtual-list` package that it was forked from. I think we should try to keep it that way, as I think that's the "common language" for scroll windowing solutions. The idea is to have VirtualList do the windowing and basically _only_ that. We could have `renderItem` and `getRowKey`, but that would turn the VirtualList component into something that knows about grids but still only does uni-directional vertical windowing.

> Cause it will keep related semantics together

I do agree with this, but I think we should do it in the opposite way, so I've pushed a commit that has us do this instead:
```js
<VirtualList role="list">
  <FileItem role="listitem" />
</VirtualList>
```
so now the FileList is fully responsible for the aria roles of the list and its items. Those two are inextricably related but previously the listitem role was hidden away into the FileItem component so that relationship wasn't clear.

I'll add comments about the presentation roles :pray: